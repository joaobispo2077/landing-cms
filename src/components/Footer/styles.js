import styled, { css } from 'styled-components';

export const Container = styled.div`
	${({ theme }) => css`
		a {
			color: inherit;
			text-decoration: none;
			font-size: ${theme.font.sizes.small};
			text-align: center;
		}
	`}
`;
