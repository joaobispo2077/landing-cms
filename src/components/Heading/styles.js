import styled from 'styled-components';

export const Title = styled.h1`
	color: ${({ theme, lighther }) =>
		lighther ? theme.colors.light : theme.colors.dark};
`;
