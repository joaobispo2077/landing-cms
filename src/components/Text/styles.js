import styled from 'styled-components';

export const Container = styled.p`
	font-size: ${({ theme }) => theme.font.sizes.medium};
`;
