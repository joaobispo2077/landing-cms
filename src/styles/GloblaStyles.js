import { createGlobalStyle } from 'styled-components';

const GloblaStyles = createGlobalStyle`
* {
	margin: 0;
	padding: 0;
}

/* body {
	background: ${({ theme }) => theme.colors.mainBg};
} */
`;

export default GloblaStyles;
