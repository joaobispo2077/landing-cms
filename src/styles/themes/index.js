export const themes = {
	default: {
		colors: {
			mainBg: 'green',
			secondaryBg: 'blue',
		},
		fonts: {},
		spacings: {},
	},
};
