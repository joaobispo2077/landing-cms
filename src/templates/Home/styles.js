import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
	${(props) => css`
		background: ${props.background};
	`};
`;
