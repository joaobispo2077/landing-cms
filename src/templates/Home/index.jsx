import * as Styled from './styles';

function Home() {
	return (
		<div>
			<Styled.Wrapper>
				<h1>Hello</h1>
			</Styled.Wrapper>
		</div>
	);
}

export default Home;
