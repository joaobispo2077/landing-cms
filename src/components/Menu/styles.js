import styled, { css } from 'styled-components';

import { Container as SectionContainer } from '../SectionContainer/styles';
import { Title as Heading } from '../Heading/styles';

export const Container = styled.div`
	${({ theme }) => css`
		position: fixed;
		z-index: 5;
		top: 0;
		right: 0;
		left: 0;

		width: 100%;

		background: ${theme.colors.light};
		border-bottom: ${theme.colors.mediumGray};

		& > ${SectionContainer} {
			padding-top: 0;
			padding-bottom: 0;
		}

		& ${Heading} {
			margin-top: 0;
			margin-bottom: 0;
		}

		@media ${theme.media.lteMedium} {
			height: 100vh;

			& > ${SectionContainer} {
				display: grid;
				grid-template-columns: 1fr;
				grid-template-rows: 1fr;

				height: 100vh;
				align-items: center;

				overflow-y: auto;
			}

			& ${Heading} {
				display: flex;
				justify-content: center;
				padding-bottom: ${theme.spacings.large};
			}
		}
	`}
`;

export const MenuContainer = styled.div`
	${({ theme }) => css`
		display: flex;
		justify-content: space-between;
		align-items: center;

		@media ${theme.media.lteMedium} {
			display: block;
			text-align: center;
			padding: ${theme.spacings.xxlarge};
		}
	`}
`;

export const Button = styled.button`
	${({ theme }) => css`
		z-index: 6;
		position: fixed;
		top: 2rem;
		right: 2rem;
		width: 4rem;
		height: 4rem;
		background: ${theme.colors.primary};
		color: ${theme.colors.light};
		cursor: pointer;
		border: none;
		display: none;
		/* transition: all 3s ease-in-out; */

		@media ${theme.media.lteMedium} {
			display: flex;
			align-items: center;
			justify-content: center;
		}

		> svg {
			width: 2.5rem;
			height: 2.5rem;
		}
	`}
`;
