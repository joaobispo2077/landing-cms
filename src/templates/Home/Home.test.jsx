import { render, screen } from '@testing-library/react';
import Home from '.';
import { themes } from '../../styles/themes';
import { renderTheme } from '../../styles/themes/renderTheme';

test('Home', () => {
	render(renderTheme(<Home />));

	const headingContainer = screen.getByRole('heading', {
		name: 'Hello',
	}).parentElement;

	expect(headingContainer).toHaveStyle({
		background: themes['default'].colors.secondaryBg,
	});

	expect(headingContainer).toHaveStyleRule('background', 'blue');

	expect(headingContainer).toMatchSnapshot();
});
