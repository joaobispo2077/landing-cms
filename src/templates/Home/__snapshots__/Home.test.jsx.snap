// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home 1`] = `
.c0 {
  background: blue;
}

<div
  class="c0"
>
  <h1>
    Hello
  </h1>
</div>
`;
